tools/release/build.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash -e
+
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script defines the automated build step of the Datalab release process.
+#
+# That involves a clean build (not using the Docker image cache) of
+# the datalab-base, datalab-gateway, and datalab images; tagging the
+# resulting images with the current date; and then pushing the
+# datalab and datalab-gateway images to the Google Container Registry.
+
+PROJECT_ID="${PROJECT_ID:-cloud-datalab}"
+TIMESTAMP=$(date +%Y%m%d)
+LABEL="${LABEL_PREFIX:-}${TIMESTAMP}"
+BACKEND_IMAGE="gcr.io/${PROJECT_ID}/datalab-gateway:${LABEL}"
```

Comment:
nit on the naming: standardize our naming for kernel gateway and datalab? Right now we have multiple names:
"backend", "kernel-gateway" for kernel gateway.
"datalab", "frondend" for datalab full (it is not really a frontend).
Maybe "kernel gateway image" and "datalab full image"? Maybe you have better names.